apps/edxapp/templates/services/nginx/configs/cms.conf.j2

```diff
@@ -6,10 +7,10 @@ server {
   listen {{ edxapp_nginx_cms_port }};
   server_name localhost;
 
-  {% if activate_http_basic_auth or edxapp_activate_http_basic_auth -%}
+{% if http_basic_auth_enabled and edxapp_nginx_cms_bypass_htaccess_ip_whitelist | length == 0 %}
   auth_basic "{{ http_basic_auth_message }}";
   auth_basic_user_file {{ http_basic_auth_user_file }};
-  {% endif %}
+{% endif %}
```

Comment:
The whole configuration is rather complex. I have the feeling that we can improve readability by nesting blocks as follow:

```jinja
{% if http_basic_auth_enabled %}
{% if bypass_htaccess %}
  location @basicauth {
    auth_basic "{{ http_basic_auth_message }}";
    auth_basic_user_file {{ http_basic_auth_user_file }};

    try_files $uri @proxy_to_cms_app;
  }
{% else %}
  auth_basic "{{ http_basic_auth_message }}";
  auth_basic_user_file {{ http_basic_auth_user_file }};
{% endif %}
{% endif %}
```

WDYT?

Reply:
:+1: